### PR TITLE
Convert custom init git fetch script to use git-sync instead

### PR DESCRIPTION
**Pull request by tim-eves**
This is both shorter and less of a potential error magnet than a custom bash script.

### CODE REVIEW

File: resources/kubectl/help-kubectl.yaml [hunk 1]
```diff
@@ -85,42 +89,32 @@ spec:
...
-          echo Configure sparse-checkout
-          echo "$GIT_SPARSE_FILES" | git sparse-checkout set --no-cone --stdin
-          git checkout
-          echo Link vendored PHP
-          ln -sf /var/www/vendor vendor
+            configMapKeyRef: {name: help-keyman-com, key: site-branch}
+        args: [
+          --repo=https://github.com/keymanapp/help.keyman.com.git,
+          --sparse-checkout-file=/tmp/git/sparse-checkout.config,
+          --one-time, --depth=1, --root=/mnt, --dest=html]
```
Comment (darcywong00) [inline resources/kubectl/help-kubectl.yaml:105]: So is `--branch=${SITE_GIT_BRANCH} \` no longer needed?

File: resources/kubectl/help-kubectl.yaml [hunk 2]
```diff
@@ -85,42 +89,32 @@ spec:
...
+          mountPath: /mnt
+        - name: git-config
+          mountPath: /tmp/git
+
+      - name: init-site-vendoring
+        image: busybox
+        securityContext:
+          runAsUser: 0
+          runAsGroup: 33
+        command: ['sh', '-c', 'ln -sf /var/www/vendor /mnt/html/vendor']
```
Comment (darcywong00) [inline resources/kubectl/help-kubectl.yaml:117]: l 44. has `mountPath: /var/www/html`. Does this link become
`ln -sf /var/www/vendor /mnt/var/html/vendor`?

File: resources/kubectl/help-kubectl.yaml (hunk 1 above)
Comment (tim-eves) [reply resources/kubectl/help-kubectl.yaml:105]: That option is for the `git` command the same info is passed via the `GIT_SYNC_BRANCH` env var in the git-sync based container, on line 99.

File: resources/kubectl/help-kubectl.yaml (hunk 2 above)
Comment (tim-eves) [reply resources/kubectl/help-kubectl.yaml:117]: The `mountPath` you mention is for the `app-php` contianer where as the `init-site-vendoring` container, where `ln` command is being run, is governed by the `mountPath: /mnt` on line 120.

**Review by darcywong00 — APPROVED** [review]
lgtm

**Pull request merged**